app/renderer/js/utils/domain-util.ts

```diff
@@ -72,6 +72,20 @@ export function updateDomain(index: number, server: ServerConf): void {
   db.push(`/domains[${index}]`, server, true);
 }
 
+export function updateDomainOrder(oldIndex: number, newIndex: number) {
+  const domains = serverConfSchema
+    .array()
+    .parse(db.getObject<unknown>("/domains"));
+
+  const [movedDomain] = domains.splice(oldIndex, 1);
+  domains.splice(newIndex, 0, movedDomain);
```

Comment:
If the database is out of sync (e.g. it has been manually edited), it’s possible for `oldIndex` and `newIndex` to be out of range. We should check for that.